R/period.R

```diff
@@ -308,21 +308,58 @@ CompleteListPeriodNames <- function(x, by)
     observed.dates <- AsDate(x, on.parse.failure = "silent")
     Period(seq(min(observed.dates), max(observed.dates), by = by), by)
 }
+
 #' \code{Period}
 #'
 #' @description Converts a date into a character.
 #' @param x The date.
-#' @param by The period used in the conversion (e.g., "week", "year").
-#' @importFrom lubridate floor_date
+#' @param by The period used in the conversion (e.g., "week", "year"). Special
+#' cases include:
+#' - "quarter", which provides the month for the quarter as YYYY-MM
+#' - "nice.quarter", which provides the quarter in the form Q1 2022
+#' - "2-week", "4-week", etc which provides multi-weekl periods labeled with
+#'   first date in the corresponding period, e.g. "2 weeks commencing 2022-07-03" 
+#'   Requires to use to specify an anchor date as a point of reference.
+#' @param anchor.date The date to use as a reference for multi-week periods. Can be a
+#' Date object, or a string in the form "YYYY-MM-DD". For example,
```

Comment:
An inconsistency, `anchor.date` is allowed to be a string or date. However, `x` must be a date. If a string is provided for `x`, then this function will error.

Reply:
made anchor.date be required as a date.